### ISSUE TITLE
yard doc/undocumented will list wrapper namespaces (Possibly intractable)

### ISSUE THREAD

**Issue by luke-hill** [lsegal/yard#1464]
## Steps to reproduce

```
module A
  module B
    # Nice doc for class
    class C
    end
  end
end
```

1. Run the following command: `yard stats --list-undoc`

## Actual Output

It flags the module `A::B` and `A` even though these aren't used...

## Expected Output

It doesn't flag the module - It "could" / "should" determine this module serves no purpose as there are no methods in it

## Environment details:

- OS: Ubuntu 18.04
- Ruby version (`ruby -v`): 3.1.1
- YARD version (`yard -v`): yard 0.9.28
- Relevant software dependency/versions: N/A

**Comment by lsegal** [comment]
> It flags the module A::B and A even though these aren't used...

It is misleading to suggest that these modules are not used. Lexically, they are certainly used when you type `A::B::C.new` or even `include A::B; C.new`. In other words, if a linter tool existed that checked for references / usage in the Ruby world, it would certainly find that A::B would be in use by any codebase that also used C, as it is necessary to "use" the outer modules to access the inner ones. 

I think you're referring to a more _conceptual_ usage in which A and B are simply "empty", or otherwise "not meant to be used"-- but neither of these situations skirt your responsibility as documenter to provide information about the modules. "Empty" is a temporal state that may change at any point in the future, and "not meant to be used" implies "private", which is all the more reason to correctly document a module as such. That said, I imagine your A/B are both public in the real-world analog of this example.

Put another way: modules should not exist for no reason. Why does A and B exist in your example? Is A a namespace for your library? Document this! (sidenote: this is a great place to put general information about your library). Is B a grouping for a specific type of class (i.e. Controllers)? Document that! This tells contributors and users alike where to look for existing objects as well as where to add these objects if new ones are created. It also informs me what I can or cannot `include` into _my own_ modules. For example, if you have a grab-bag of A::Utils, I may choose to `include A::Utils` into my namespaces to make usage lexically simpler, even if Utils itself contains no direct methods.

If it helps, know that for repeated namespace-type modules, you only need to document them one time in your codebase, not in each individual file. Adding `# Controllers contains all request controllers in the A package` in any file would suffice.

More importantly though: YARD does not care about "usage" when it comes to `yard stats`. It's not YARD's job to determine what is or isn't _used_-- the command is there to simply inform you of which objects are missing documentation. YARD would need a full generalized learning algorithm behind it in order to determine which of these warnings could be discarded on your behalf, and even then it would still be wrong a good chunk of the time (since even humans cannot actually predict usage). 

tl;dr this is expected behavior. Even if YARD based its checks on usage, both A and B are used (in the lexical sense) anywhere C would be and thus are part of the exposed API.
